modules/reactor/src/meshgenerators/CoreMeshGenerator.C

```diff
@@ -354,81 +391,118 @@ std::unique_ptr<MeshBase>
+      // from pin_type and region id from assembly_type_id, z_id, and peripheral_index
+      dof_id_type assembly_type_id = elem->get_extra_integer(assembly_type_id_int);
+      unsigned int peripheral_idx = (UINT16_MAX - 1) - pin_type_id;
+      bool is_background_region = peripheral_idx == 0;
+      const auto elem_rid =
+          (is_background_region ? _background_region_id_map[assembly_type_id][z_id]
+                                : _duct_region_id_map[assembly_type_id][z_id][peripheral_idx - 1]);
+      elem->set_extra_integer(region_id_int, elem_rid);
+
+      // Set element block name and block id
+      auto elem_block_name = default_block_name;
+      if (is_background_region)
       {
-        // element is in an assembly duct or background region since it doesn't have a pin type id
-        // that matches one in the map. What region it is in the periphery is then stored in the
-        // pt_id
-        dof_id_type at_id = elem->get_extra_integer(atid_int);
-        unsigned int peripheral_index = (UINT16_MAX - 1) - pt_id;
-        if (peripheral_index == 0)
-          // background region
-          elem->subdomain_id() = _background_id_map.find(at_id)->second[z_id];
-        else
-          // duct region
-          elem->subdomain_id() = _duct_id_map.find(at_id)->second[z_id][peripheral_index - 1];
+        bool has_background_block_name = !_background_block_name_map[assembly_type_id].empty();
+        if (has_background_block_name)
+          elem_block_name += "_" + _background_block_name_map[assembly_type_id][z_id];
       }
       else
       {
-        // region is in a pin so grab the different axial region ids and swap them
-        // since during extrusion all regions are given the same ID as the 2D layer
-        for (const auto i : make_range((_id_map.at(pt_id))[0].size()))
-        {
-          // swap subdomain region ids if they are different
-          if (r_id == _id_map.at(pt_id)[0][i] &&
-              _id_map.at(pt_id)[0][i] != _id_map.at(pt_id)[z_id][i])
-            elem->subdomain_id() = _id_map.at(pt_id)[z_id][i];
-        }
+        bool has_duct_block_names = !_duct_block_name_map[assembly_type_id].empty();
+        if (has_duct_block_names)
+          elem_block_name += "_" + _duct_block_name_map[assembly_type_id][z_id][peripheral_idx - 1];
       }
+      if (elem->type() == TRI3 || elem->type() == PRISM6)
+        elem_block_name += "_TRI";
+      updateElementBlockNameId(
+          *(*_build_mesh), elem, rgmb_name_id_map, elem_block_name, next_block_id);
     }
   }
-  else
-  {
-    _mesh_name = _empty_pos ? name() + "_deleted" : name() + "_pattern";
-  }
 
-  std::string region_id_name = "region_id";
-  if (!(*_build_mesh)->has_elem_integer(region_id_name))
-  {
-    mooseError("Expected mesh inputs to have the extra integer id: region_id.\n");
-  }
+  // Sideset 10000 does not get stitched properly when BlockDeletionGenerator
```

Comment:
to what is it stitched exactly?
is this just the outer_core sideset ?

Reply:
Typically in Reactor MG's sideset 10000 is reserved for the outer boundary of the final Reactor mesh. In the RGMB CoreMeshGenerator, however, this sideset does not get generated properly if there are multiple dummy assemblies, since we are supposed to generate independent dummy assemblies for each dummy assembly position but instead we only define one dummy assembly and populate it into multiple locations. This block of code, therefore, just copies the "outer_core" sideset (which is generated properly) into sideset 10000